src/app/services/baw-api/analysis/analysis-job-item-results.service.ts

```diff
@@ -0,0 +1,84 @@
+import { Injectable } from "@angular/core";
+import {
+  emptyParam,
+  filterParam,
+  IdOr,
+  id,
+  IdParamOptional,
+  option,
+  ReadonlyApi,
+  param,
+} from "@baw-api/api-common";
+import { BawApiService, Filters } from "@baw-api/baw-api.service";
+import { stringTemplate } from "@helpers/stringTemplate/stringTemplate";
+import { AnalysisJobItemResults } from "@models/AnalysisJobItemResults";
+import { AnalysisJob } from "@models/AnalysisJob";
+import { Observable } from "rxjs";
+import { AudioRecording } from "@models/AudioRecording";
+import { Resolvers } from "@baw-api/resolver-common";
+
+const analysisJobId: IdParamOptional<AnalysisJob> = id;
+const audioRecordingId: IdParamOptional<AudioRecording> = id;
+const analysisJobItemResultsPath = param;
+
+// Example of results endpoint:
+// /analysis_jobs/{analysisJob}/results/{audioRecordingId}/Towsey.Acoustic/folderA/fileA.wav
+const endpoint = stringTemplate`/analysis_jobs/${analysisJobId}/results/${audioRecordingId}${analysisJobItemResultsPath}${option}`;
+
+@Injectable()
+export class AnalysisJobItemResultsService
+  implements
+    ReadonlyApi<
+      AnalysisJobItemResults,
+      [IdOr<AnalysisJob>, IdOr<AudioRecording>]
+    >
+{
+  public constructor(private api: BawApiService<AnalysisJobItemResults>) {}
+
+  public list(
+    analysisJob: IdOr<AnalysisJob>,
+    audioRecording: IdOr<AudioRecording>
+  ): Observable<AnalysisJobItemResults[]> {
+    return this.api.list(
+      AnalysisJobItemResults,
+      endpoint(analysisJob, audioRecording, emptyParam, emptyParam)
+    );
+  }
+
+  public filter(
+    filters: Filters<AnalysisJobItemResults>,
+    analysisJob: IdOr<AnalysisJob>,
+    audioRecording: IdOr<AudioRecording>
+  ): Observable<AnalysisJobItemResults[]> {
+    return this.api.filter(
+      AnalysisJobItemResults,
+      endpoint(analysisJob, audioRecording, emptyParam, filterParam),
+      filters
+    );
+  }
+
+  public show(
+    analysisJobItemResult: AnalysisJobItemResults,
+    analysisJob: IdOr<AnalysisJob>,
+    audioRecording: IdOr<AudioRecording>
+  ): Observable<AnalysisJobItemResults> {
```

Comment:
this does not appear to accept a path parameter

Reply:
The path parameter is passed as part of the `analysisJobItemResult` model. This is inline with how the path parameter is passed in the harvest services.

Would it be more correct to explicitly accept the path as a parameter?

Reply:
I've added `TODO:` code comments to
> consider overloads that take a path or model

Reply:
Yeah we won't always have a model, we'll just have a path. Let's tackle this later when we need to.